File: spring-data-requery-java/src/test/java/com/coupang/springframework/data/requery/java/domain/basic/BasicEntityTest.java
package com.coupang.springframework.data.requery.java.domain.basic;

import com.coupang.springframework.data.requery.java.domain.AbstractDomainTest;
import org.junit.Test;

import static org.assertj.core.api.Assertions.assertThat;

/**
 * com.coupang.springframework.data.requery.java.domain.basic.BasicEntityTest
 *
 * @author debop
 * @since 18. 6. 4
 */
public class BasicEntityTest extends AbstractDomainTest {


    @Test
    public void insertUser() throws InterruptedException {
        BasicUser user = new BasicUser();
        user.setName("test");

        requeryTemplate.upsert(user);

        BasicUser loaded = requeryTemplate.findById(BasicUser.class, user.id);

        assertThat(loaded.getLastModifiedDate()).isNull();
        loaded.name = "updated";
        requeryTemplate.update(loaded);

        assertThat(loaded.getLastModifiedDate()).isNotNull();

        Thread.sleep(100L);
    }
}

File: spring-data-requery-kotlin/src/test/kotlin/org/springframework/data/requery/kotlin/domain/basic/BasicEntityTest.kt
package org.springframework.data.requery.kotlin.domain.basic

import org.assertj.core.api.Assertions.assertThat
import org.junit.Before
import org.junit.Test
import org.springframework.data.requery.kotlin.domain.AbstractDomainTest
import org.springframework.data.requery.kotlin.domain.RandomData

class BasicEntityTest: AbstractDomainTest() {

    @Before
    fun setup() {
        operations.deleteAll(BasicLocationEntity::class)
        operations.deleteAll(BasicGroupEntity::class)
        operations.deleteAll(BasicUserEntity::class)
    }

    @Test
    fun `insert user without association`() {

        val user = RandomData.randomBasicUser()
        operations.insert(user)

        val loaded = operations.findById(BasicUserEntity::class, user.id)!!

        loaded.name = "updated"
        operations.update(loaded)

        assertThat(loaded.name).isEqualTo("updated")
    }

    @Test
    fun `select specific columns with limit`() {
        val user = RandomData.randomBasicUser()
        operations.insert(user)

        val result = operations
            .select(BasicUserEntity::class, BasicUserEntity.ID, BasicUserEntity.NAME)
            .limit(10)
            .get()

        val first = result.firstOrNull()
        assertThat(first).isNotNull
        assertThat(first.id).isEqualTo(user.id)
        assertThat(first.name).isEqualTo(user.name)

        // NOTE: Lazy loading 을 수행합니다.
        assertThat(first.age).isEqualTo(user.age)
    }
}